gengokit/generator/gen_test.go

```diff
@@ -346,82 +340,79 @@ func TestUpdateMethods(t *testing.T) {
 		PBPackage: "github.com/TuneLab/go-truss/gengokit/general-service",
 	}
 
-	te, err := gengokit.NewTemplateExecutor(sd, conf)
+	te, err := gengokit.NewExecutor(sd, conf)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	testHandlerGeneration := func(templPath string) {
-		svc.Methods = []*svcdef.ServiceMethod{allMethods[0]}
-		firstBytes, err := testGenerateResponseFile(templPath, te, nil)
-		if err != nil {
-			t.Fatal(err)
-		}
-		firstCode := strings.TrimSpace(string(firstBytes))
+	templPath := handler.ServerFile
 
-		secondCode, err := renderService(svc, firstCode, te, templPath)
-		if err != nil {
-			t.Fatal(err)
-		}
+	svc.Methods = []*svcdef.ServiceMethod{allMethods[0]}
 
-		if strings.Compare(firstCode, secondCode) != 0 {
-			t.Fatal("Generated code differs after regenerated with same definition\n" +
-				templPath + "\n" +
-				diff(firstCode, secondCode))
-		}
+	firstCode, err := renderService(svc, "", te, templPath)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		svc.Methods = append(svc.Methods, allMethods[1])
+	secondCode, err := renderService(svc, firstCode, te, templPath)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		thirdCode, err := renderService(svc, secondCode, te, templPath)
-		if err != nil {
-			t.Fatal(err)
-		}
+	if strings.Compare(firstCode, secondCode) != 0 {
+		t.Fatal("Generated code differs after regenerated with same definition\n" +
+			templPath + "\n" +
+			diff(firstCode, secondCode))
+	}
 
-		if strings.Compare(secondCode, thirdCode) != -1 {
-			t.Fatal("Generated code not longer after regenerated with additional service method\n" +
-				templPath + "\n" +
-				diff(secondCode, thirdCode))
-		}
+	svc.Methods = append(svc.Methods, allMethods[1])
 
-		// remove the first one rpc
-		svc.Methods = svc.Methods[1:]
+	thirdCode, err := renderService(svc, secondCode, te, templPath)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		forthCode, err := renderService(svc, thirdCode, te, templPath)
-		if err != nil {
-			t.Fatal(err)
-		}
+	if thirdCode <= secondCode {
```

Comment:
This will practically never affect this code path, but this test is not a reliable method for determining the length of a string, [as illustrated here](https://play.golang.org/p/Tm45ji81iv). If you want to test the length of a string, use `len(str)`.